integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/v2/NextMoveValidator.java

```diff
@@ -0,0 +1,144 @@
+
+        String standard = sbd.getStandard();
+
+        if (!documentType.fitsDocumentIdentifier(standard)) {
+            throw new DocumentTypeDoNotFitDocumentStandardException(documentType, standard);
+        }
+
+        if (!serviceRecord.hasStandard(standard)) {
+            throw new ReceiverDoNotAcceptDocumentStandard(standard, sbd.getProcess());
+        }
+
+        Class<?> group = ValidationGroupFactory.toServiceIdentifier(serviceIdentifier);
+        asserter.isValid(sbd.getAny(), group != null ? new Class<?>[]{group} : new Class<?>[0]);
+    }
+
+    void validate(NextMoveOutMessage message) {
+        // Must always be at least one attachment
+        StandardBusinessDocument sbd = message.getSbd();
+        if (!sbdUtil.isReceipt(sbd) && (message.getFiles() == null || message.getFiles().isEmpty())) {
+            throw new MissingFileException();
+        }
+
+        sbd.getExpectedResponseDateTime().ifPresent(expectedResponseDateTime -> {
+            if (sbdUtil.isExpired(sbd)) {
+                timeToLiveHelper.registerErrorStatusAndMessage(sbd, message.getServiceIdentifier(), message.getDirection());
+                throw new TimeToLiveException(expectedResponseDateTime);
+            }
+        });
+
+        if (sbdUtil.isType(message.getSbd(), ARKIVMELDING)) {
+            // Verify each file referenced in arkivmelding is uploaded
+            List<String> arkivmeldingFiles = ArkivmeldingUtil.getFilenames(getArkivmelding(message));
+            Set<String> messageFiles = message.getFiles().stream()
+                    .map(BusinessMessageFile::getFilename)
+                    .collect(Collectors.toSet());
+
+            List<String> missingFiles = arkivmeldingFiles.stream()
+                    .filter(p -> !messageFiles.contains(p))
+                    .collect(Collectors.toList());
+
+            if (!missingFiles.isEmpty()) {
+                throw new MissingArkivmeldingFileException(String.join(",", missingFiles));
+            }
+        }
+
+        if (message.getServiceIdentifier() == DPI && message.getFiles().stream()
+                .noneMatch(BusinessMessageFile::getPrimaryDocument)) {
+            throw new MissingPrimaryDocumentException();
+        }
+    }
+
+    private Arkivmelding getArkivmelding(NextMoveOutMessage message) {
+        // Arkivmelding must exist for DPO
+        BusinessMessageFile arkivmeldingFile = message.getFiles().stream()
+                .filter(f -> NextMoveConsts.ARKIVMELDING_FILE.equals(f.getFilename()))
+                .findAny()
+                .orElseThrow(MissingArkivmeldingException::new);
+
+        try (FileEntryStream fileEntryStream = cryptoMessagePersister.readStream(message.getMessageId(), arkivmeldingFile.getIdentifier())) {
+            return ArkivmeldingUtil.unmarshalArkivmelding(fileEntryStream.getInputStream());
+        } catch (JAXBException | IOException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)